slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala

```diff
@@ -230,4 +230,90 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
       }
     } yield ()).withPinnedSession
   }
+
+  // TODO DerbyProfile and PostgresProfile seems to have a problem in this case. check #2206 and #2207
+  def testInsertOrUpdateWithPrimaryKeyOnly: DBIOAction[Unit, NoStream, Effect.All] = if (!tdb.confName.matches("(derby|postgres).*")) {
+    class T(tag: Tag) extends Table[Int](tag, "mytable") {
+      def id = column[Int]("id", O.PrimaryKey)
+      def * = id
+    }
+    case class V(id: Int, value: String)
+    class T2(tag: Tag) extends Table[V](tag, "mytable2") {
+      def id = column[Int]("id")
+      def value = column[String]("value", O.SqlType("varchar(20)"))
```

Comment:
Why hardcode a O.SqlType when there's a O.Length for specifying string length in database-aware manner?